src/deluge/processing/engines/audio_engine.cpp

```diff
@@ -1411,11 +1412,15 @@ void slowRoutine() {
 	     - SSI_TX_BUFFER_NUM_SAMPLES)
 	    & (SSI_RX_BUFFER_NUM_SAMPLES - 1);
 
-	if (latencyWithinAppropriateWindow >= SSI_TX_BUFFER_NUM_SAMPLES) {
+	while (latencyWithinAppropriateWindow >= SSI_TX_BUFFER_NUM_SAMPLES) {
 		i2sRXBufferPos += (SSI_TX_BUFFER_NUM_SAMPLES << (2 + NUM_MONO_INPUT_CHANNELS_MAGNITUDE));
 		if (i2sRXBufferPos >= (uint32_t)getRxBufferEnd()) {
 			i2sRXBufferPos -= (SSI_RX_BUFFER_NUM_SAMPLES << (2 + NUM_MONO_INPUT_CHANNELS_MAGNITUDE));
 		}
+		latencyWithinAppropriateWindow =
+		    (((rxBufferWriteAddr - (uint32_t)i2sRXBufferPos) >> (2 + NUM_MONO_INPUT_CHANNELS_MAGNITUDE))
+		     - SSI_TX_BUFFER_NUM_SAMPLES)
+		    & (SSI_RX_BUFFER_NUM_SAMPLES - 1);
```

Comment:
Am I understanding correctly that this is a busy wait until the I2S ADC gives us enough samples to slam in to the record buffer? The loop could probably benefit from an explanatory comment.

Reply:
It's just a bug fix in the existing correction code. If the rx buffer was in the 256-384 range it would correct it one receive buffer of samples, but if it was outside that range to start it would still be wrong.  

Replacing the if with a while loop adds 128 until it's in the 256-384 range relative to the tx buffer since it occasionally lands in the 2000ish range. I think it's caused by rendering getting ahead of the desired (per Rohan's comment) 128 sample latency